cleanlab/object_detection/summary.py

```diff
@@ -444,6 +449,138 @@ def visualize(
     plt.show()
 
 
+def _get_per_class_confusion_matrix_dict_(
+    labels: List[Dict[str, Any]],
+    predictions: List[np.ndarray],
+    iou_threshold: Optional[float] = 0.5,
+    num_procs: int = 1,
+) -> DefaultDict[int, Dict[str, int]]:
+    """
+    Returns a confusion matrix dictionary for each class containing TP,FP and FN.
+    """
+    num_classes = len(predictions[0])
+    num_images = len(predictions)
+    pool = Pool(num_procs)
+    counter_dict: DefaultDict[int, collections.Counter[int]] = collections.defaultdict(
+        collections.Counter
+    )
+
+    for class_num in range(num_classes):
+        pred_bboxes, lab_bboxes = _filter_by_class(labels, predictions, class_num)
+        tpfp = pool.starmap(
+            _calculate_true_positives_false_positives,
+            zip(
+                pred_bboxes,
+                lab_bboxes,
+                [iou_threshold for _ in range(num_images)],
+                [True for _ in range(num_images)],
+            ),
+        )
+        for j, tpfp_j in enumerate(tpfp):
+            for k, tpfp_k in enumerate(tpfp_j):
+                counter_dict[class_num][k] += np.sum(tpfp_k)
+
+    results: DefaultDict[int, Dict[str, int]] = collections.defaultdict(dict)
+    for class_num in counter_dict:
+        results[class_num]["TP"] = counter_dict[class_num][0]
+        results[class_num]["FP"] = counter_dict[class_num][1]
+        results[class_num]["FN"] = counter_dict[class_num][2]
+    return results
+
+
+def get_average_per_class_confusion_matrix_(
+    labels: List[Dict[str, Any]], predictions: List[np.ndarray], num_procs: int = 1
+) -> Dict[int, Dict[str, float]]:
+    """
```

Comment:
missing one-line of what this method is supposed to do. This is a public-facing method and needs clear description of its purpose

Reply:
you need to make it clear that a high-level this is about evaluating the trained object detection model.

Most of the methods in cleanlab are about assessing the data itself. So when we have methods for assessing models, it needs to be clearly stated this method is for assessing models, not data